lstmcpipe/stages/onsite_mc_merge_and_copy_dl1.py

```diff
@@ -143,22 +143,14 @@ def merge_dl1(input_dir, particle2jobs_dict, particle=None, flag_merge=False, fl
     if not len(os.listdir(DL1_training_dir)) == len(read_lines_file(training_filelist)):
         tf = check_files_in_dir_from_file(DL1_training_dir, training_filelist)
         if tf != []:
-            # TODO to fix this
-            print(f"\n\n\n ****!!!!! {len(tf)} files from the training list are not in the `DL1/training directory:\n{tf} "
-                  "Continuing\n\n\n")
-        # if tf != [] and not flag_full_workflow:
-        #     query_continue("{} files from the training list are not in the `DL1/training` directory:\n{} "
-        #                    "Continue ?".format(len(tf), tf))
+            raise Exception(f" * {len(tf)} files from the training list are not in the `DL1/training directory:\n{tf}"
+                  "\nContinuing.")
```

Comment:
You are not really continuing if you raise an Exception, right?

Reply:
It was just printing the warning before, so this is a change in behaviour.
is that intended?

Reply:
No intended indeed...^^ thanks for the catch
How would you proceed (also as an user) @LukasNickel ? If the tests (checking that all the files that were processed are indeed gonna be merged) doesn't pass, would you stop the pipeline or would you just raise a warning (and in the clos future store it in the log ?)

Reply:
As a user I would probably suspect, that the pipeline stops at that point. I could then debug the issue and once it is resolved, continue with my analysis.
If the output looks alright, nobody is going to look into the logs for a while, so nobody might notice that there is an error in the files.

If its just one broken file, it would be nice if the chain could continue where it left off, once that has been fixed. But that might be a bit tricky (and maybe the separation into datalevels is enough for that)

Reply:
> As a user I would probably suspect, that the pipeline stops at that point. I could then debug the issue and once it is resolved, continue with my analysis.
The pipeline can only stops when it gets to this point (stages will take time to finish). But this messages will appear before because the filelist are written when the pipeline is launched although the files are not processed yet. 


> If its just one broken file, it would be nice if the chain could continue where it left off, once that has been fixed. But that might be a bit tricky (and maybe the separation into datalevels is enough for that)

This is quite difficult to do indeed... (I think 😄 ). I don't know how to put some jobs in "standby" inside a cluster if they fail  and re-launch them later. At least we will need to think how to do it very carefully.

Reply:
Wait, the message appears before?
It should process files, then make sure all files are processed and then print a warning or fail, right?
Of course it cant fail before it processes the dl1 stage, why woul dthe message be printed before?

Reply:
Hmm.. i'm indeed not sure what this part is really doing (it should not work as the sub-dl1 files are not processed and created yet, so this check should always fail...)

Another point in the todo list ;-) improve the merge stage 😬

Reply:
OK, I have to recognise that i'm not sure how this check was not triggering before.... it will always trigger indeed. I think that I would just take it out 'cause it will never be satisfied, indeed.

Reply:
And do some tests with master version in case I'm missing/messing something big

Reply:
Again, confused for all the changes (and doing 3 things at the same time...^^).

The original code was 
```
   if tf != [] and not flag_full_workflow:
```

so of course this check was not happening. Now that there is no `flag_full_workflow` so this check will always 'fail'. Taking out the check then.

Thanks for the catch @LukasNickel